csrc/cpu/cpu_types.hpp

```diff
@@ -336,14 +478,35 @@ template <> inline void storeFP32<c10::BFloat16>(float v, c10::BFloat16 *ptr) {
   *ptr = *(v_ptr + 1);
 }
 
+#ifdef __AVX512F__
 inline BF16Vec8::BF16Vec8(const FP32Vec8 &v)
     : reg(_mm256_cvtepi32_epi16(
           _mm256_bsrli_epi128(_mm256_castps_si256(v.reg), 2))) {}
 
 inline BF16Vec16::BF16Vec16(const FP32Vec16 &v)
     : reg(_mm512_cvtepi32_epi16(
           _mm512_bsrli_epi128(_mm512_castps_si512(v.reg), 2))) {}
-#endif
+#else
+namespace{
+__m128i FP32Vec8_to_BF16Vec8_avx2(__m256 a) {
+  __m256i ai = _mm256_castps_si256(a);
+  ai = _mm256_srli_epi32(ai, 16);
+  ai = _mm256_packus_epi32(ai, ai);
+  ai = _mm256_permute4x64_epi64(ai, 0b00111001);
+  return _mm256_extracti128_si256(ai, 0);
```

Comment:
Maybe we can use ```_mm256_cvtepi32_epi16``` directly.

Reply:
> Maybe we can use `_mm256_cvtepi32_epi16` directly.

It can't be used with AVX2 ISA, right?
https://www.intel.com/content/www/us/en/docs/intrinsics-guide/index.html#text=_mm256_cvtepi32_epi16&ig_expand=2958,1619
<img width="728" alt="image" src="https://github.com/vllm-project/vllm/assets/19923746/675733ff-0deb-49d0-9d8a-859be1021e8e">